python/lsst/pipe/drivers/multiBandDriver.py

```diff
@@ -160,7 +162,29 @@ def __init__(self, butler=None, schema=None, refObjLoader=None, reuse=tuple(), *
         self.reuse = tuple(reuse)
         self.makeSubtask("detectCoaddSources")
         self.makeSubtask("mergeCoaddDetections", schema=schema)
-        self.makeSubtask("measureCoaddSources", schema=afwTable.Schema(self.mergeCoaddDetections.schema),
+        if self.config.measureCoaddSources.inputCatalog.startswith("deblended"):
+            # Ensure that the output from deblendCoaddSources matches the input to measureCoaddSources
+            self.measurementInput = self.config.measureCoaddSources.inputCatalog
+            self.deblenderOutput = []
+            if self.config.deblendCoaddSources.simultaneous:
+                if self.config.deblendCoaddSources.multiBandDeblend.conserveFlux:
+                    self.deblenderOutput.append("deblendedFlux")
+                if self.config.deblendCoaddSources.multiBandDeblend.saveTemplates:
+                    self.deblenderOutput.append("deblendedModel")
+            else:
+                self.deblenderOutput.append("deblendedFlux")
+            if self.measurementInput not in self.deblenderOutput:
+                err = "Measurement input '{0}' is not in the list of deblender output catalogs '{1}'"
+                raise ValueError(err.format(self.measurementInput, self.deblenderOutput))
```

Comment:
Do lines 166-178 do more than checking if things are configured correctly? The comment suggested so to me. If so, this whole block can practically be removed, right?  I'm a bit uneasy such checking lives in `pipe_drivers`; it looks like introducing additional logics beyond stitching tasks into a workflow.

Reply:
You are correct that they only check whether or not the output from the deblender is consistent with the input to the measurements. I feel like it's important to check that the tasks are consistent before running them, otherwise the user would have to wait for the entire deblending task to complete before crashing, and the error might not be obvious.

Reply:
Could this validation be moved to `MultiBandDriverConfig.validate`?